src/plugin/peripheral/nodes/base-node.ts

```diff
@@ -55,8 +55,8 @@ export abstract class PeripheralBaseNode extends BaseTreeNode {
         this.pinned = false;
     }
 
-    public selected(): Thenable<boolean> {
-        return Promise.resolve(false);
+    public async selected(): Promise<boolean> {
```

Comment:
Why `async`? Wouldn't the `Promise` return value enough here without any awaited promises inside the method? And wouldn't `async` on overrides by enough when needed?

Reply:
I'm not sure I fully understand what you are asking but basically when we use the async method (which requires us to return a Promise) we can avoid creating the promise in the body as it is automatically created for us. Any error thrown within the method body will result in a rejected promise.

Reply:
Should have started this with a "nit".
I think in this case you get away without the `async` qualifier because you do not `await` a promise. The important part is to change the return type to `Promise<boolean>`. But I might be wrong. Best to ignore this comment. Nothing's really wrong here.